contracts/cw1-subkeys/src/contract.rs

```diff
@@ -365,14 +354,14 @@ fn can_send<S: Storage, A: Api, Q: Querier>(
             }
         }
         CosmosMsg::Staking(staking_msg) => {
-            permissions_read(&deps.storage)
-                .may_load(owner_raw.as_slice())
-                .map(|perm_opt| {
-                    // if permission exists
-                    perm_opt.map_or(false, |perm| {
-                        check_staking_permissions(&staking_msg, perm).is_ok()
-                    })
-                })
+            let perm_opt = permissions_read(&deps.storage).may_load(owner_raw.as_slice())?;
+            match perm_opt {
+                Some(permission) => match check_staking_permissions(&staking_msg, permission) {
```

Comment:
This could actually be:

`Some(permission) => Ok(check_staking_permissions(&staking_msg, permission).is_ok())`

Error only returned from `may_load` unable to parse.